lib/dispatcher.js

```diff
@@ -4,7 +4,12 @@ Router.prototype.dispatcher = function () {
   var trie = this.trie
 
   return function* trieRouter (next) {
-    var match = trie.match(this.request.path)
+    try {
+      var match = trie.match(this.request.path)
+    } catch (err) {
+      err.code = 1101;
```

Comment:
also, you're not going to get an error in the client - only at the server. you should send a response from the server and test that response from the client